File: Java/src/test/java/myservice/MyServiceTest.java
package myservice;

import static org.junit.Assert.*;

import org.junit.Test;

import sso.Request;
import sso.Response;
import sso.SSOToken;
import sso.SingleSignOnRegistry;

public class MyServiceTest {

    @Test
    public void validSSOTokenIsAccepted() {
        // TODO: replace registry with a Test Double
        SingleSignOnRegistry registry = null;
        MyService service = new MyService(registry);
        SSOToken token = new SSOToken();
        Response response = service.handleRequest(new Request("Foo", token));
        assertEquals("hello Foo!", response.getText());
    }
}

File: CSharp/UnitTests/MyServiceTest.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using SingleSignOnKata.myservice;
using SingleSignOnKata.sso;
using Moq;

namespace UnitTests
{
    [TestClass]
    public class MyServiceTest
    {
        [TestMethod]
        public void ValidSSOTokenIsAccepted()
        {
            // TODO: use a test double for the SingleSignOnRegistry
            SingleSignOnRegistry registry = null;
            MyService service = new MyService(registry);
            SSOToken token = new SSOToken();
            Response response = service.HandleRequest(new Request("Foo", token));
            Assert.AreEqual("hello Foo!", response.GetText());
        }
    }
}
